quarkus/service/src/main/java/org/apache/polaris/service/quarkus/config/QuarkusProducers.java

```diff
@@ -100,13 +100,8 @@ public PolarisDiagnostics polarisDiagnostics() {
 
   @Produces
   @RequestScoped
-  public RealmId realmId(@Context HttpServerRequest request, RealmIdResolver realmIdResolver) {
-    return realmIdResolver.resolveRealmContext(
-        request.absoluteURI(),
-        request.method().name(),
-        request.path(),
-        request.headers().entries().stream()
-            .collect(HashMap::new, (m, e) -> m.put(e.getKey(), e.getValue()), HashMap::putAll));
+  public RealmId realmId(@Context ContainerRequestContext request) {
+    return (RealmId) request.getProperty(RealmIdFilter.REALM_ID_KEY);
```

Comment:
Should we check that the property is present (as it should at this stage)?... just for extra safety.

Reply:
This is implicitly the case already: if the property is absent, this method returns null – and returning null from a producer method is forbidden:

https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#producer_method

Reply:
TIL - thx :sweat_smile: